File: src/main/kotlin/com/github/pikokr/chino/structs/Listener.kt
package com.github.pikokr.chino.structs

import net.dv8tion.jda.api.events.GenericEvent
import kotlin.reflect.KClass

abstract class Listener<T: GenericEvent> {
    val type: KClass<T>

    constructor(type: KClass<T>) {
        this.type = type
    }

    open fun execute(event: T) {}
}

File: src/util/listener/Listener.ts
import ChinoClient from "../ChinoClient";

export default class Listener {
    bot: ChinoClient
    event: string
    __path: string
    emitter: string

    constructor(client: ChinoClient, emitter: string, event: string) {
        this.bot = client
        this.event = event
        this.emitter = emitter
        this.__path = ''
    }

    execute(...args: any[]) {}
}
